kernel/src/common/interrupt.c

```diff
@@ -322,12 +299,15 @@ void enable_interrupt(InterruptType mask) {
         case BOTH:
             asm volatile("cpsie if");
             break;
+        default:
+            /** should never happen **/
```

Comment:
Maybe panic here?

Reply:
no the default should go. it switches on an enum so when we implement all enum cases there is no default

Reply:
it should be there after all (clion otherwise complains). panic indeed

Reply:
Enabling interrupts with an incorrect mask is weird but not something that harms our state. So a panic isn't warranted I think.

Reply:
I think we need at least a warn there. but a fatal isn't that bad either. It's an enum. If you really pass it some random integer something should start to scream